### PR TITLE
use lang from api for addon fields in html

### LINKED ISSUE #3049: Declare language for user localized text
### Describe the problem and steps to reproduce it:

* Browse the site in your default locale (such as French)
* Go an add-on detail page where the default locale of the description is something other than French (say, English)

### What happened?

The section of the text in English does not have a `lang` attribute which means it will be interpreted as French (or whatever locale is in the URL).

### What did you expect to happen?

It should have a `lang` attribute specifying the locale.

### Anything else we should know?

* This is important for [screen readers](http://adrianroselli.com/2015/01/on-use-of-lang-attribute.html).
* The old site did it.
* We should probably also include the `dir` attribute for right-to-left languages but I don't know what the practical usage of that is.
* This should apply to all user localized data, such as reviews, version release notes, etc.

### PR DESCRIPTION
**Pull request by GregSherman**
Fixes #3049 

Create a new type LocalizedStringWithLocale and new function SelectLocalizedContentWithLocale so that the AddonType (internal Addon) fields that previously use SelectLocalizedContent now contains a 'locale' field. Everywhere in the code where any of these 5 fields (summary, name, description, support_email, developer_comments) exist, the surrounding HTML will contain a lang tag (if applicable).

### CODE REVIEW

**Comment by GregSherman** [comment]
It's entirely possible I missed areas to put a lang tag inside, I can make further edits if there do exist some

**Comment by GregSherman** [comment]
@eviljeff @diox Requesting review

File: src/amo/components/RatingManager/index.js [hunk 1]
```diff
@@ -206,6 +208,7 @@ export class RatingManagerBase extends React.Component<InternalProps> {
           <legend
             className="RatingManager-legend"
             dangerouslySetInnerHTML={promptHTML}
+            lang={addon.name.locale}
```
[the remark this answers is not in the document]
Comment (eviljeff) [reply src/amo/components/RatingManager/index.js:211]: Oh, and this should be tested.

File: src/amo/components/SearchResult/index.js [hunk 2]
```diff
@@ -164,7 +164,10 @@ export class SearchResultBase extends React.Component<InternalProps> {
     if (showSummary) {
       const summaryProps = {};
       if (addon) {
-        summaryProps.dangerouslySetInnerHTML = sanitizeHTML(addon.summary);
+        summaryProps.dangerouslySetInnerHTML = sanitizeHTML(
+          addon.summary ? addon.summary.content : '',
+        );
+        summaryProps.lang = addon.summary ? addon.summary.locale : null;
```
Comment (eviljeff) [inline src/amo/components/SearchResult/index.js:170]: This should be tested.

File: src/amo/pages/Addon/index.js [hunk 3]
```diff
@@ -330,6 +336,7 @@ export class AddonBase extends React.Component {
         <div
           className="Addon-developer-comments-contents"
           dangerouslySetInnerHTML={devComments}
+          lang={addon.developer_comments.locale}
```
Comment (eviljeff) [inline src/amo/pages/Addon/index.js:339]: This should be tested.

File: src/amo/pages/Addon/index.js [hunk 4]
```diff
@@ -429,13 +436,18 @@ export class AddonBase extends React.Component {
...
+        : addon.description.content;
+      const summaryLang = addon.summary
+        ? addon.summary.locale
+        : addon.description.locale;
       if (summary && summary.length) {
         summaryProps.dangerouslySetInnerHTML = sanitizeHTML(nl2br(summary), [
           'a',
           'br',
         ]);
+        summaryProps.lang = summaryLang;
```
Comment (eviljeff) [inline src/amo/pages/Addon/index.js:450]: This should be tested.

File: src/amo/reducers/utils.js [hunk 5]
```diff
@@ -17,3 +17,25 @@ export const selectLocalizedContent = (
 
   return field[lang];
 };
+
+export const selectLocalizedContentWithLocale = (
```
Comment (eviljeff) [inline src/amo/reducers/utils.js:21]: We need a test for this function

File: src/amo/reducers/addons.js [hunk 6]
```diff
@@ -230,8 +233,11 @@ export function createInternalAddon(
     review_url: apiAddon.review_url,
     slug: apiAddon.slug,
     status: apiAddon.status,
-    summary: selectLocalizedContent(apiAddon.summary, lang),
-    support_email: selectLocalizedContent(apiAddon.support_email, lang),
+    summary: selectLocalizedContentWithLocale(apiAddon.summary, lang),
+    support_email: selectLocalizedContentWithLocale(
+      apiAddon.support_email,
+      lang,
+    ),
```
Comment (eviljeff) [inline src/amo/reducers/addons.js:240]: Add to the tests in reducers/test_addons.js make sure lang is set correctly too

File: src/blog-utils/StaticAddonCard/index.js [hunk 7]
```diff
@@ -65,6 +75,7 @@ export const StaticAddonCardBase = ({
 
       <div className="StaticAddonCard-summary">
         <p
+          lang={summaryLang}
```
Comment (eviljeff) [inline src/blog-utils/StaticAddonCard/index.js:78]: test for this too

**Review by eviljeff — CHANGES_REQUESTED** [review]
Your changes seem fine overall, there just needs to be some extra asserts added to the tests to check that `lang` is being added to the markup everywhere it should be.  (I tried to comment each place I noticed it was added).

And it's worth finding and understanding where `selectLocalizedContent` is still used to make sure we've not missed something.

File: src/amo/types/api.js [hunk 8]
```diff
@@ -22,6 +22,11 @@ export type LocalizedString = {
   [locale: string]: string,
 };
 
+export type LocalizedStringWithLocale = {
```
[the remark this answers is not in the document]
Comment (GregSherman) [reply src/amo/types/api.js:25]: LocalizedString is used throughout the codebase with many collection types, version licenses, ExternalLanguageToolType, LocalizedUrlWithOutgoing, and some more that refer to some addon attributes as localizedstring such as ExternalBlockType. I suppose anywhere that is using "selectLocalizedContent" can be changed to use selectLocalizedContentWithLocale and add it to whichever html tag. Though they are not always related to addon names (like ExternalLanguageToolType)

[the remark this answers is not in the document]
Comment (eviljeff) [reply src/amo/types/api.js:25]: It does sound like `LocalizedStringWithLocale` could be used anywhere the API returns a localized string then, but we don't want to increase the scope of this issue/pr.  Lets leave them for a possible follow-up issue.

**Comment by GregSherman** [comment]
If this is good to go, I'll make another issue for the remaining selectLocalizedContent areas

File: src/amo/components/SearchResult/index.js (hunk 2 above)
Comment (eviljeff) [reply src/amo/components/SearchResult/index.js:170]: I couldn't see the test for this - can you point to it (or add it)?

**Review by eviljeff — APPROVED** [review]
Looks good to me.  Just a question on the test I couldn't find.

File: src/amo/components/SearchResult/index.js (hunk 2 above)
Comment (eviljeff) [reply src/amo/components/SearchResult/index.js:170]: (it's confusing that the test isn't in a file matching the name of the component, but...) I meant the case when `summary` isn't set and the code is doing something slightly different.

Comment (GregSherman) [reply src/amo/components/SearchResult/index.js:170]: I see, when the summary doesn't exist, there was no test for it. I added it in the same file below the test in the most recent commit

**Comment by eviljeff** [comment]
(same deal about needing to comment in the issue to have it assigned to you)

**Pull request merged**